controller.go

```diff
@@ -778,6 +778,8 @@ func (c *controller) addNetwork(n *network) error {
 		return err
 	}
 
+	n.startResolver()
+
```

Comment:
@msabansal Can a container be connected to only one network in Windows ? Even if that is the case today for nat networks I think we should consider the possibility in future a container can be connected to multiple networks. Hence the resolver instance should be associated with the sandbox and not per network. We already have platform specific files, `sandbox_dns_unix.go` and `sandbox_dns_windows.go` to abstract out the implementation details of `startResolver`.

The name/ip lookup functions are not platform specific and the service DB `svcRecords` is already keyed by the network ID. So the lookup functions `ResolveName` etc. can remain in sandbox.go and doesn't have to be refactored.